dockers/docker-lldp-sv2/lldpmgrd

```diff
@@ -0,0 +1,193 @@
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+
+def log_warning(msg):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_WARNING, msg)
+    syslog.closelog()
+
+
+def log_error(msg):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+
+# ========================== Signal Handling ==========================
+
+def signal_handler(sig, frame):
+    if sig == signal.SIGHUP:
+        log_info("Caught SIGHUP - ignoring...")
+        return
+    elif sig == signal.SIGINT:
+        log_info("Caught SIGINT - exiting...")
+        sys.exit(128 + sig)
+    elif sig == signal.SIGTERM:
+        log_info("Caught SIGTERM - exiting...")
+        sys.exit(128 + sig)
+    else:
+        log_warning("Caught unhandled signal '" + sig + "'")
+
+
+# ============================== Classes ==============================
+
+class LldpManager(object):
+    """
+    Class which subscribes to notifications of changes in the PORT table of
+    the Redis State database and updates LLDP configuration accordingly for
+    that port by calling lldpcli.
+    Attributes:
+        state_db: Handle to Redis State database via swsscommon lib
+        config_db: Handle to Redis Config database via swsscommon lib
+    """
+    REDIS_HOSTNAME = "localhost"
+    REDIS_PORT = 6379
+    REDIS_TIMEOUT_USECS = 0
+
+    def __init__(self):
+        # Open a handle to the State database
+        self.state_db = swsscommon.DBConnector(swsscommon.STATE_DB,
+                                               self.REDIS_HOSTNAME,
+                                               self.REDIS_PORT,
+                                               self.REDIS_TIMEOUT_USECS)
+
+        # Open a handle to the Config database
+        self.config_db = swsscommon.DBConnector(swsscommon.CONFIG_DB,
+                                                self.REDIS_HOSTNAME,
+                                                self.REDIS_PORT,
+                                                self.REDIS_TIMEOUT_USECS)
+
+    def update_lldp_config_for_port(self, port_name):
+        """
+        For port `port_name`, look up the neighboring device's hostname and
+        corresponding port alias in the Config database, then form the
+        appropriate lldpcli configuration command and run it.
+        """
+        CMD_FMT = "lldpcli configure ports {0} lldp portidsubtype local {1} description {2}:{3}"
+        TABLE_SEPARATOR = "|"
+
+        # Retrieve all entires from the Device Neighbor table
+        device_neighbor_table = swsscommon.Table(self.config_db, swsscommon.CFG_DEVICE_NEIGHBOR_TABLE_NAME, TABLE_SEPARATOR)
+        (status, fvp) = device_neighbor_table.get(port_name)
+        if not status:
+            log_info("Port '{}' not found in {} table in Config DB. Not updating LLDP config for this port.".format(port_name, swsscommon.CFG_DEVICE_NEIGHBOR_TABLE_NAME))
+            return
```

Comment:
to be complete, we should also listen to CFG_DEVICE_NEIGHBOR_TABLE_NAME change, since if user configure this table, we may need to update the message we send out. I understand this is not an issue now, but we should add an comment to mark as TODO.

Reply:
I do not think we should return here, in case there is no description, we should send "ldpcli configure ports {0} lldp portidsubtype local {1}". this will ensure we send out the correct port name instead of mac address.